helix-term/src/commands.rs

```diff
@@ -2315,10 +2315,25 @@ enum Operation {
     Change,
 }
 
+fn only_whole_lines(selection: &Selection, text: &Rope) -> bool {
+    selection.ranges().iter().all(|range| {
+        // If not at least a full line is selected (strange require 2).
+        if range.slice(text.slice(..)).len_lines() < 2 {
+            return false;
+        }
+        // If the start of the selection is at the start of a line and the end at the end of a line.
+        let (start_line, end_line) = range.line_range(text.slice(..));
+        let start = text.line_to_char(start_line);
+        let end = text.line_to_char((end_line + 1).min(text.len_lines()));
+        start == range.from() && end == range.to()
```

Comment:
There are a few places here that turn the rope into a ropeslice. We can slice once at the beginning instead: `let text = text.slice(..);`